### ISSUE TITLE
[RFC] Naming Conventions

### ISSUE THREAD

**Issue by JaredWright** [Bareflank/hypervisor#903]
This RFC shows two possible naming conventions that could be used to define interfaces on the "solid_foundation" branch, Right now, they are using "Option 1" below, but it would take almost no effort to switch everything to "Option 2". After writing out the examples below, I actually think I'm leaning toward switching to "Option 2" for *everything* including PAL functions. What are other people's opinions? @connojd @rianquinn @brendank310?

## Option 1: Use the pattern (noun)_(verb)
Pros:
* Visually groups together calls within an interface
* Immediately suggests which interface the call belongs to
* Consistent with how we've named intrinsics in the past

Cons:
* Doesn't read like english
* You have to keep reading all the way to the end of the line before you know which action the function is performing

Example:
```
void my_vcpu_init(x64_vcpu &vcpu) noexcept
{
    vcpu.io_port_vmexit_handler_set(my_vmexit_handle);
    vcpu.io_port_vmexit_enable(0x10);
    vcpu.io_port_vmexit_range_enable(0xCF8, 0xCFC);
    vcpu.io_port_vmexit_range_enable(0xFF0, 0xFFF);

    vcpu.rdmsr_vmexit_handler_set(my_vmexit_handle);
    vcpu.rdmsr_vmexit_enable(0x1000);
    vcpu.rdmsr_vmexit_range_enable(0x2000, 0x4000);

    pal::ia32_efer_sce_enable();
    pal::ia32_efer_lme_enable();
    pal::ia32_efer_lma_enable();
    pal::ia32_efer_nxe_enable();

    pal::vm_exit_controls_save_debug_controls_enable();
    pal::vm_exit_controls_host_address_space_size_enable();
    pal::vm_exit_controls_save_ia32_pat_enable();
    pal::vm_exit_controls_load_ia32_pat_enable();

    return;
}
```

## Option 2: Use the pattern (verb)_(noun)
Pros:
* Reads like english, very intuitive to read from left to right like a sentence
* Immediately suggests the action of the function

Cons:
* Calls within the same interface don't visually line up
* Doesn't immediately suggest which interface the call belongs to

Example:
```
void init_my_vcpu(x64_vcpu &vcpu) noexcept
{
    vcpu.set_io_port_vmexit_handler(handle_my_vmexit);
    vcpu.enable_io_port_vmexit(0x10);
    vcpu.enable_io_port_vmexit_range(0xCF8, 0xCFC);
    vcpu.enable_io_port_vmexit_range(0xFF0, 0xFFF);

    vcpu.set_rdmsr_vmexit_handler(handle_my_vmexit);
    vcpu.enable_rdmsr_vmexit(0x1000);
    vcpu.enable_rdmsr_vmexit_range(0x2000, 0x4000);

    pal::enable_ia32_efer_sce();
    pal::enable_ia32_efer_lme();
    pal::enable_ia32_efer_lma();
    pal::enable_ia32_efer_nxe();

    pal::enable_vm_exit_controls_save_debug_controls();
    pal::enable_vm_exit_controls_host_address_space_size();
    pal::enable_vm_exit_controls_save_ia32_pat();
    pal::enable_vm_exit_controls_load_ia32_pat();

    return;
}
```

**Comment by connojd** [comment]
I vote for option 2

**Comment by rianquinn** [comment]
I am used to option #1 and I think I originally wrote everything that way for the same reasons that are listed in option #1, but I think my vote is for option #2. It definitely reads better.

**Comment by JaredWright** [comment]
3/4 Maintainers voted for option 2, so I made the switch. This changed is merged into solid_foundation in PR #904.
